src/datreant/core/collections.py

```diff
@@ -148,26 +148,29 @@ def __getitem__(self, index):
         select out members.
 
         """
+        memberlist = self._list()
         # we can take lists of indices, names, or uuids; these return a
         # View; repeats already not respected since View functions as a
         # set
-        if ((isinstance(index, list) or hasattr(index, 'dtype')) and
-                all([isinstance(item, bool) for item in index])):
+        if ((isinstance(index, list) and
+             all(isinstance(item, bool) for item in index)
+             or
+             hasattr(index, 'dtype') and index.dtype == 'bool')):
```

Comment:
How are the bools supposed to be evaluated? Does `or` have a greater precedence then `and`? To make sure you should use brackets to properly assign the evaluations. Also it would be good to make the `hasattr` check first to make use of lazy evaluation for the and checks.